### ISSUE TITLE
[feature request] https support

### ISSUE THREAD

**Issue by mueslimak3r** [proconsule/nxmp#1]
support for ssl & tls exists in libnx. It would be awesome to use this to support https streams in NXMP!

**Comment by proconsule** [comment]
Sure is in my TODO list

**Comment by ITotalJustice** [comment]
todo this, you have to enable tls protocal in ffmpeg and link against mbedtls.

```
  ./configure --prefix=$PORTLIBS_PREFIX --disable-shared --enable-static \
    --cross-prefix=aarch64-none-elf- --enable-cross-compile \
    --arch=aarch64 --target-os=horizon --enable-pic \
    --extra-cflags='-D__SWITCH__ -D_GNU_SOURCE -O2 -march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIC -ftls-model=local-exec' \
    --extra-cxxflags='-D__SWITCH__ -D_GNU_SOURCE -O2 -march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIC -ftls-model=local-exec' \
    --extra-ldflags='-fPIE -L${PORTLIBS_PREFIX}/lib -L${DEVKITPRO}/libnx/lib' \
    --disable-runtime-cpudetect --disable-programs --disable-debug --disable-doc --disable-autodetect \
    --disable-avdevice --disable-hwaccels --disable-encoders \
    --enable-swscale --enable-swresample --enable-network \
    --disable-protocols --enable-protocol=file,http,ftp,tcp,udp,rtmp,hls,tls \
    --enable-zlib --enable-bzlib --enable-libass --enable-libfreetype --enable-libfribidi \
    --enable-mbedtls --enable-version3 --enable-demuxer=hls
 ```
 
NOTE: there was a bug in ffmpeg with mbedtls which did not close fd, so it would crash on switch after 15 or so opens, i fixed this upstream awhile ago. im not sure if the version in dkp is still the old version, but you can just change to latest stable (4.41) and the ffmpeg.patch will still work.

also, building will currently fail as libnx has `inet_aton` already in header which ffmpeg has its own version due to `HAVE_INET_ATON` not being defined. you can patch this though.

**Comment by Ch0cola** [comment]
Available on 0.5.0

**Comment by mueslimak3r** [comment]
Awesome! To use this would I use `source=TestHTTP|https://192.168.1.10/path/` or `source=TestHTTPS|https://192.168.1.10/path/` ?
